Fix RPC Python User Function Error Handling

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#63406 Fix RPC Python User Function Error Handling**

The `RemoteException` will be thrown on the caller side when converting
the response message to IValue. Since it is a Python error, the error
message needs to be extracted explicitly and clear the `PyErr`.

Differential Revision: [D30372741](https://our.internmc.facebook.com/intern/diff/D30372741)

mrshenli committed:
Fix RPC Python User Function Error Handling

The `RemoteException` will be thrown on the caller side when converting
the response message to IValue. Since it is a Python error, the error
message needs to be extracted explicitly and clear the `PyErr`.

[ghstack-poisoned]

mrshenli commented:
@mrshenli has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30372741).

mrshenli committed:
Update on "Fix RPC Python User Function Error Handling"


The `RemoteException` will be thrown on the caller side when converting
the response message to IValue. Since it is a Python error, the error
message needs to be extracted explicitly and clear the `PyErr`.

Differential Revision: [D30372741](https://our.internmc.facebook.com/intern/diff/D30372741)

[ghstack-poisoned]

mrshenli commented:
@mrshenli has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30372741).

rohan-varma reviewed (approved):
Looks like some tests fail but otherwise looks good.

mrshenli committed:
Update on "Fix RPC Python User Function Error Handling"


The `RemoteException` will be thrown on the caller side when converting
the response message to IValue. Since it is a Python error, the error
message needs to be extracted explicitly and clear the `PyErr`.

Differential Revision: [D30372741](https://our.internmc.facebook.com/intern/diff/D30372741)

[ghstack-poisoned]

mrshenli commented:
@mrshenli has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30372741).

mrshenli committed:
Update on "Fix RPC Python User Function Error Handling"


The `RemoteException` will be thrown on the caller side when converting
the response message to IValue. Since it is a Python error, the error
message needs to be extracted explicitly and clear the `PyErr`.

Differential Revision: [D30372741](https://our.internmc.facebook.com/intern/diff/D30372741)

[ghstack-poisoned]

mrshenli commented:
@mrshenli has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30372741).

mrshenli committed:
Update on "Fix RPC Python User Function Error Handling"


The `RemoteException` will be thrown on the caller side when converting
the response message to IValue. Since it is a Python error, the error
message needs to be extracted explicitly and clear the `PyErr`.

Differential Revision: [D30372741](https://our.internmc.facebook.com/intern/diff/D30372741)

[ghstack-poisoned]

mrshenli commented:
@mrshenli has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30372741).
